Source/engine/animationinfo.h

```diff
@@ -37,41 +37,46 @@ enum AnimationDistributionFlags : uint8_t {
  */
 class AnimationInfo {
 public:
-	/**
-	 * @brief Animation sprite
-	 */
-	std::optional<CelSprite> celSprite;
-	/**
-	 * @brief How many game ticks are needed to advance one Animation Frame
-	 */
-	int TicksPerFrame;
-	/**
-	 * @brief Increases by one each game tick, counting how close we are to TicksPerFrame
-	 */
-	int TickCounterOfCurrentFrame;
-	/**
-	 * @brief Number of frames in current animation
-	 */
-	int NumberOfFrames;
-	/**
-	 * @brief Current frame of animation
-	 */
-	int CurrentFrame;
-	/**
-	 * @brief Is the animation currently petrified and shouldn't advance with gfProgressToNextGameTick
-	 */
-	bool IsPetrified;
+	// clang-format off
+	// generic getters
+	[[nodiscard]] const std::optional<CelSprite>  &getCelSprite() const { return celSprite_; }
+	[[nodiscard]] int getCurrentFrame()                           const { return currentFrame_; }
+	[[nodiscard]] int getNumberOfFrames()                         const { return numberOfFrames_; }
+	[[nodiscard]] int getTickCounterOfCurrentFrame()              const { return tickCounterOfCurrentFrame_; }
+	[[nodiscard]] int getTicksPerFrame()                          const { return ticksPerFrame_; }
+	[[nodiscard]] int getTriggerFrame()                           const { return triggerFrame_; }
+
+	// helper getters
+	[[nodiscard]] bool isLastFrame()         const { return (currentFrame_ == numberOfFrames_ - 1); }
+	[[nodiscard]] bool isOver()              const { return (currentFrame_ >= numberOfFrames_ - 1); }
+	[[nodiscard]] bool isTriggerFrame()      const { return (currentFrame_ == triggerFrame_); }
+	[[nodiscard]] bool isAfterTriggerFrame() const { return (currentFrame_  > triggerFrame_); }
+
+	// generic setters
+	void setCelSprite(const std::optional<CelSprite> &celSprite) { celSprite_ = celSprite; }
+	void setCurrentFrame(const int frameNumber)                  { currentFrame_ = frameNumber; }
+	void setNumberOfFrames(const int numberOfFrames)             { numberOfFrames_ = numberOfFrames; }
+	void setTickCounterOfCurrentFrame(const int tickCounter)     { tickCounterOfCurrentFrame_ = tickCounter; }
+	void setTicksPerFrame(const int ticksPerFrame)               { ticksPerFrame_ = ticksPerFrame; } 
+	void setTriggerFrame(const int triggerFrame)                 { triggerFrame_ = triggerFrame; }
+
+	// helper setters
+	void emplaceCelSprite(const OwnedCelSprite &celSprite)  { celSprite_.emplace(celSprite); }
+	void setCurrentFrameAsLast()                            { currentFrame_ = numberOfFrames_ - 1; }

```

Comment:
fairly awkward name for this function, something like `seekLastFrame`, `activateLastFrame`? a matching `seekFirstFrame`/`activateFirstFrame` function would be useful too.

Reply:
`seekLastFrame` sounds awkward to me As it would be good for some function that returns it. I could agree to `activateLastFrame`... but maybe there will be even better propositions, like you know activate is like set something to active disactivate - non active state...? Actually `setCurrentFrameToLast` maybe? leaves not doubt for me. What are other options?

Reply:
seek was based on the convention for file I/O methods, but there's probably something more suitable in the context of an animation sequence.